app/code/Magento/Inventory/Indexer/StockItem/IndexDataProvider.php

```diff
@@ -37,9 +39,18 @@ public function __construct(ResourceConnection $resourceConnection)
     public function getData(int $stockId, array $sourceIds = []): \ArrayIterator
     {
         $connection = $this->resourceConnection->getConnection();
+        $sourceTable = $connection->getTableName(SourceResourceModel::TABLE_NAME_SOURCE);
         $sourceItemTable = $connection->getTableName(SourceItemResourceModel::TABLE_NAME_SOURCE_ITEM);
         $sourceStockLinkTable = $connection->getTableName(StockSourceLinkResourceModel::TABLE_NAME_STOCK_SOURCE_LINK);
 
+        // find all enabled sources
+        $select = $connection->select()->from($sourceTable, [Source::SOURCE_ID])->where(Source::ENABLED . '=?', 1);
+        if (count($sourceIds)) {
+            $select->where(Source::SOURCE_ID . ' = ?', $sourceIds);
+        }
+        $sourceIds = $connection->fetchCol($select);
```

Comment:
Need to check this condition
If we do not have enabled sources then we need to return empty data without extra queries